pumas_stochastic_collect_tau.F90

```diff
@@ -0,0 +1,810 @@
+
+integer, parameter, public  :: ncd = 35
+integer, parameter, public  :: ncdp = ncd + 1
+integer, parameter, public  :: ncdl = ncd
+integer, parameter, public  :: ncdpl = ncdl+1
+
+!integer, private :: ncd,ncdp
+!integer, private :: ncdl,ncdpl
+!PARAMETER(ncd=35,ncdl=ncd) ! set number of ice and liquid bins
+!PARAMETER(ncdp=ncd+1,ncdpl=ncdl+1)
+
+! for Zach's collision-coalescence code
+
+
+real(r8), private :: knn(ncd,ncd)
+
+real(r8), public :: mmean(ncd), diammean(ncd)       ! kg & m at bin mid-points
+real(r8), public :: medge(ncdp), diamedge(ncdp)     ! kg & m at bin edges 
+integer, private  :: cutoff_id                       ! cutoff between cloud water and rain drop, D = 40 microns
+
+! Assume 6 microns for each...
+real(r8), parameter :: m1 = 4._r8/3._r8*pi*rhow*(6.e-6_r8)**3
+
+
+!===============================================================================
+contains
+!===============================================================================
+
+      
+subroutine calc_bins    
+
+real(r8) :: DIAM(ncdp)
+real(r8) :: X(ncdp)
+real(r8) :: radsl(ncdp)
+real(r8) :: radl(ncd)
+integer  :: L, lcl  
+real(r8) :: kkfac
+!Then before doing any calculations you'll need to calculate the bin mass grid 
+! (note this code could be cleaned up, I'm just taking it as it's used in our bin scheme). 
+! This only needs to be done once, since we'll use the same bin mass grid for all calculations. 
+
+! use mass doubling bins from Graham Feingold (note cgs units)
+
+!      PI=3.14159_r8
+!      rho_w=1000._r8                         ! kg/m3
+      DIAM(1)=1.5625*2.E-04_r8                ! cm
+      X(1)=PI/6._r8*DIAM(1)**3*rhow/1000._r8  ! rhow kg/m3 --> g/cm3 
+      radsl(1) = X(1)                         ! grams 
+!      radsl(1) = X(1)/1000._r8       
+
+      DO l=2,ncdp
+         X(l)=2._r8*X(l-1)
+         DIAM(l)=(6._r8/pi*X(l)*1000._r8/rhow)**(1._r8/3._r8)  ! cm
+!         radsl(l)=X(l)/1000._r8 ! convert from g to kg
+         radsl(l)=X(l)             
+      ENDDO
+
+! now get bin mid-points
+
+      do l=1,ncd
+         radl(l)=(radsl(l)+radsl(l+1))/2._r8         ! grams   
+!         diammean(l) = (DIAM(l)+DIAM(l+1))/2._r8     ! cm
+         diammean(l) = (6._r8/pi*radl(l)*1000._r8/rhow)**(1._r8/3._r8) ! cm
+      end do
+
+! set bin grid for method of moments
+
+        ! for method of moments
+
+      do lcl = 1,ncd+1
+!         medge(lcl) = radsl(lcl)*1000._r8     ! convert to grams
+         medge(lcl) = radsl(lcl)               ! grams
+         diamedge(lcl) = DIAM(lcl)             ! cm
+      enddo
+
+      do lcl = 1,ncd
+!         mmean(lcl) = radl(lcl)*1000._r8
+         mmean(lcl) = radl(lcl)  
+         diammean(lcl) = diammean(lcl)
+      enddo
+
+      do lcl = ncdp,1,-1
+         if( diamedge(lcl).ge.40.e-4_r8 ) cutoff_id = lcl
+      end do  
+
+      write(*,*) 'cutoff_id = ', cutoff_id
+         
+end subroutine calc_bins
+
+subroutine pumas_stochastic_kernel_init(iunit)
```

Comment:
My version (the one in cam_dev) of stochastic_tau_cam.F90, has a namelist which reads in the filename and passes the filename into this routine to be opened and read.

Reply:
My modification which impacts this line has not been committed yet.  I will resolve this conversation.